source/contribute.rst

```diff
@@ -267,76 +347,67 @@ sense to revert a commit documenting a feature without also reverting
+Rebasing
+--------
 
-Next, merge or rebase your commit against **doctrine/2.11.x**. With your
-work done in a **local** topic branch, you'll want to assist upstream
-merge by rebasing your commits. You can either do this manually with
-``fetch`` then ``rebase``, or use the ``pull --rebase`` shortcut. You
-may encounter merge conflicts, which you should fix and then mark as
-fixed with ``add``, and then continue rebasing with
-``rebase --continue``. At any stage, you can abort the rebase with
-``rebase --abort`` unlike nasty merges which will leave files strewn
-everywhere.
+On upstream changes
+~~~~~~~~~~~~~~~~~~~
 
-.. code-block:: console
+Sometimes, you will need to rebase your branch on the latest changes,
+typically because the build had an issue unrelated to your changes, and
+that issue has been fixed after you created your branch.
 
-    $ git fetch doctrine
-    $ git rebase doctrine/2.11.x fix-weird-bug
-
-Push your branch to **origin**:
-
-Finished topic branches should be pushed to **origin** for a
-**maintainer** to review and pull into **doctrine** as appropriate:
-
-.. code-block:: console
-
-    $ git push origin fix-weird-bug
-    To git@github.com:hobodave/dbal.git
-        * [new branch]      fix-weird-bug -> fix-weird-bug
-
-Now you are ready to send a pull request from this branch and ask for a
-review from a maintainer.
+Basically, a rebase takes all changes on your topic branch and moves
+them to another starting point. This starting point was the Doctrine
+branch that you chose when you created your topic branch, at that point
+in time. The rebase will move your changes to be based on the current
+state of this branch.
 
-Topic Branch Cleanup
---------------------
-
-Once your work has been merged by the branch maintainer, it will no
-longer be necessary to keep the local branch or remote branch, so you
-can remove them!
+Here is how to proceed if you need to rebase on ``2.3.x``:
 
-Sync your local 2.11.x branch:
+1. Fetch all new commits: `git fetch doctrine``.
+2. Rebase on what you fetched:
+   ``git rebase doctrine/2.3.x``
+3. If you run into a conflict, fix it with ``git mergetool``, then
+   continue on your merry way with ``git rebase --continue``.
+4. Force push to overwrite the previous version : ``git push --force``.
 
-.. code-block:: console
+.. _rebasing-on-another-branch:
 
-    $ git checkout 2.11.x
-    $ git pull --rebase
+On another branch
+~~~~~~~~~~~~~~~~~
 
-Remove your local topic branch using -d to ensure that it has been merged by
-upstream. Branch -d will not delete a branch that is not an ancestor of
-your current head.
+Another case where you need a rebase is when you want to change the
+target branch of your PR. For instance, you might have created your PR
+against ``2.3.x`` but you are told to change it to ``2.4.x``. In that
+case, if ``2.4.x`` does not have all the commits that ``2.3.x`` has, you
+need to drop these commits while rebasing: it is not your responsibility
+to merge them up.
```

Comment:
Good point, I was not aware of that!

Reply:
Wait, subconsciously I was.

```
git checkout my-topic
git rebase doctrine/2.3.x --onto doctrine/2.4.x
```

No need to pick the right commits. Am I missing something?

Reply:
You're probably right, let's just suggest that one method.

Reply:
When using that method, the "commits you did not author" thingy is not a issue at all, right? If so, we need not mention it.

---

Another case where you need a rebase is when you want to change the
target branch of your PR. For instance, you might have created your PR
against ``2.3.x`` but you are told to change it to ``2.4.x``. In that
case, the following command will pick all changes that you made against the ``2.3.x`` branch, and re-apply them on the current ``2.4.x`` branch.

$ git fetch doctrine
$ git rebase --onto doctrine/2.4.x doctrine/2.3.x your-topic-branch
$ git push --force

Reply:
> When using that method, the "commits you did not author" thingy is not a issue at all, right? If so, we need not mention it.

Not an issue indeed 👍